xarray/core/duck_array_ops.py

```diff
@@ -715,12 +709,16 @@ def mean(array, axis=None, skipna=None, **kwargs):
     if dtypes.is_datetime_like(array.dtype):
         offset = _datetime_nanmin(array)
 
-        # xarray always uses np.datetime64[ns] for np.datetime64 data
-        dtype = "timedelta64[ns]"
+        # From version 2025.01.2 xarray uses np.datetime64[unit], where unit
+        # is one of "s", "ms", "us", "ns".
+        # To not have to worry about the resolution, we just convert the output
+        # to "int64" and let the dtype of offset take precedence.
+        # This is fully backwards compatible with datetime64[ns]. For other
+        # this might lead to imprecise output, where fractional parts are truncated.
         return (
             _mean(
                 datetime_to_numeric(array, offset), axis=axis, skipna=skipna, **kwargs
-            ).astype(dtype)
+            ).astype("int64")
```

Comment:
```suggestion
            ).astype("timedelta64")
```

Reply:
OK, casting from `float64` to `int64` introduces `RuntimeWarning: invalid value encountered in cast` in presence of `NaT`. The result is nevertheless correct. We can avoid that, by casting to `"m8"` (without unit).